src/common/src/types/interval.rs

```diff
@@ -73,6 +73,23 @@ impl Interval {
         }
     }
 
+    /// Creates a new `Interval` from the given number of years, months, days, hours and
+    /// microseconds.
+    pub fn from_year_month_day_hour_minute_usec(
```

Comment:
This func only used in `str_to_interval`, why not just do the calculation inside `str_to_interval` and call `from_month_day_usec`?